h2/utilities.py

```diff
@@ -319,13 +317,82 @@ def _check_host_authority_header(headers, hdr_validation_flags):
     # an :authority header nor a Host header.
     if not authority_present and not host_present:
         raise ProtocolError(
-            "Did not receive an :authority or Host header."
+            "Request header block must have an :authority or Host header."
         )
 
     # If we receive both headers, they should definitely match.
     if authority_present and host_present:
         if authority_header_val != host_header_val:
             raise ProtocolError(
-                "Received mismatched :authority and Host headers: %r / %r" %
-                (authority_header_val, host_header_val)
+                "Request header block must have matching :authority and "
+                "Host headers: %r / %r"
```

Comment:
Let's change this to a statement of what is wrong: `"Request header block has mismatched :authority and Host headers"`